### PR TITLE
SQL variables: add a new variable `tidb_enable_parallel_apply`

### PR DESCRIPTION
**Pull request by qw4990**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

**Comment by TomShawn** [comment]
@qw4990 Could you involve a tech review?

**Comment by qw4990** [comment]
PTAL @SunRunAway

File: system-variables.md [hunk 1]
```diff
@@ -956,3 +956,9 @@ explain select * from t where age=5;
 +-------------------------------+------------+-----------+-----------------------------+-------------------------------+
 3 rows in set (0.00 sec)
 ```
+
+### `tidb_enable_parallel_apply`
```
Comment (SunRunAway) [inline system-variables.md:960]: Apply 并不是一个用户知道的用词。有没有更好的词？

**Comment by TomShawn** [comment]
@qw4990 This PR has been held for several months without any update. Shall we close this PR?

File: system-variables.md [hunk 2]
```diff
@@ -1136,6 +1136,13 @@ set tidb_slow_log_threshold = 200;
 - 默认值：ON
 - 这个变量用于控制计算窗口函数时是否采用高精度模式。
 
+
+### `tidb_enable_parallel_apply`
```
Comment (TomShawn) [inline system-variables.md:1140]:
```suggestion
### `tidb_enable_parallel_apply` <span class="version-mark">从 v5.0 GA 版本开始引入</span>
```

File: system-variables.md [hunk 3]
```diff
@@ -1136,6 +1136,13 @@ set tidb_slow_log_threshold = 200;
...
+
+- 作用域：SESSION | GLOBAL
+- 默认值：1
+- 这个变量用于控制是否开启并行 Apply 执行，并发数由 tidb_executor_concurrency 控制。
```
Comment (TomShawn) [inline system-variables.md:1144]: 能否说明一下什么是并行 Apply 执行呢（是并行执行 Apply 算子吗）？开启这个功能有什么好处呢？目前默认开启情况如何，是否有需要注意的地方呢？

File: system-variables.md [hunk 4]
```diff
@@ -1136,6 +1136,13 @@ set tidb_slow_log_threshold = 200;
 - 默认值：ON
 - 这个变量用于控制计算窗口函数时是否采用高精度模式。
 
+
```
Comment (TomShawn) [inline system-variables.md:1139]:
```suggestion
```

File: system-variables.md [hunk 5]
```diff
@@ -1136,6 +1136,12 @@ set tidb_slow_log_threshold = 200;
 - 默认值：ON
 - 这个变量用于控制计算窗口函数时是否采用高精度模式。
 
+### `tidb_enable_parallel_apply` <span class="version-mark">从 v5.0 GA 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 默认值：0
+- 这个变量用于控制是否开启 Apply 并发，并发数由 tidb_executor_concurrency 控制。Apply 算子用来处理关联子查询，且默认无并发所以速度较慢，此时可打开这个开关，增加并发度，提高执行速度。
```
Comment (TomShawn) [inline system-variables.md:1143]:
```suggestion
- 这个变量用于控制是否开启 Apply 并发，并发数由 tidb_executor_concurrency 控制。Apply 算子用来处理关联子查询且默认无并发，所以执行速度较慢。打开 Apply 并发开关可增加并发度，提高执行速度。目前默认关闭。
```

**Comment by qiancai** [comment]
@Joyinqin The doc changes in this PR are ready for EN PR creation. PTAL, thanks~

**Pull request merged**